lib/agenda.js

```diff
@@ -200,33 +200,33 @@ Agenda.prototype.every = function(interval, names, data) {
   }
 };
 
-Agenda.prototype.schedule = function(when, names, data) {
+Agenda.prototype.schedule = function(when, names, data, lastcb) {
   var self = this;
 
   if (typeof names === 'string' || names instanceof String) {
-    return createJob(when, names, data);
+    return createJob(when, names, data, lastcb);
   } else if (Array.isArray(names)) {
-    return createJobs(when, names, data);
+    return createJobs(when, names, data, lastcb);
   }
 
-  function createJob(when, name, data) {
+  function createJob(when, name, data, cb) {
     var job = self.create(name, data);
     job.schedule(when);
-    job.save();
+    job.save(cb);
     return job;
   }
 
-  function createJobs(when, names, data) {
+  function createJobs(when, names, data, cb) {
     return names.map(function (name) {
-      return createJob(when, name, data);
+      return createJob(when, name, data, cb);
```

Comment:
this is a bug and will cause cb to be called multiple times.

need to use `async.map` or similar